File: benchmarks/fixture/ruby/billing.rb
module FixtureBilling
  def self.calculate_invoice_total(lines, tax_rate)
    subtotal = lines.sum { |line| line[:quantity] * line[:unit_price] }
    (subtotal * (1 + tax_rate)).round(2)
  end

  def self.validate_payment(amount, currency)
    amount.positive? && currency.length == 3
  end
end

File: benchmarks/fixture/python/billing.py
"""Billing helpers for the benchmark fixture."""


def calculate_invoice_total(lines: list[dict[str, float]], tax_rate: float) -> float:
    subtotal = sum(line["quantity"] * line["unit_price"] for line in lines)
    return round(subtotal * (1.0 + tax_rate), 2)


def validate_payment(amount: float, currency: str) -> bool:
    return amount > 0 and len(currency) == 3
